examples/decomposition/plot_faces_decomposition.py

```diff
@@ -120,57 +146,62 @@ def plot_gallery(title, images, n_col=n_col, n_row=n_row, cmap=plt.cm.gray):
 ]
 
 
-# #############################################################################
-# Plot a sample of the input data
-
-plot_gallery("First centered Olivetti faces", faces_centered[:n_components])
-
-# #############################################################################
-# Do the estimation and plot it
+# %%
+# Next, fit each estimator on all images and plot some results.
+# Each estimator extracts 6 components as vectors
+# :math:`h \in \mathbf{R}^{4096}`. We just displayed these vectors
+# in human-friendly visualisation as 64x64 pixel images.
+#
+# .. note::
+#
+#     The Eigenfaces estimator, via the :py:mod:`sklearn.decomposition.PCA`,
+#     also provides a scalar `noise_variance_` (the mean of pixelwise variance)
+#     that cannot be displayed as an image.
 
 for name, estimator, center in estimators:
-    print("Extracting the top %d %s..." % (n_components, name))
+    print("- Extracting the top %d '%s'..." % (n_components, name))
     t0 = time()
-    data = faces
     if center:
         data = faces_centered
+    else:
+        data = faces
     estimator.fit(data)
     train_time = time() - t0
-    print("done in %0.3fs" % train_time)
+    print("  done in %0.3fs" % train_time)
     if hasattr(estimator, "cluster_centers_"):
         components_ = estimator.cluster_centers_
     else:
         components_ = estimator.components_
 
-    # Plot an image representing the pixelwise variance provided by the
-    # estimator e.g its noise_variance_ attribute. The Eigenfaces estimator,
-    # via the PCA decomposition, also provides a scalar noise_variance_
-    # (the mean of pixelwise variance) that cannot be displayed as an image
-    # so we skip it.
     if (
         hasattr(estimator, "noise_variance_") and estimator.noise_variance_.ndim > 0
-    ):  # Skip the Eigenfaces case
+    ):  # skip the Eigenfaces case
         plot_gallery(
-            "Pixelwise variance",
+            "Pixelwise variance from \n %s" % name,
             estimator.noise_variance_.reshape(1, -1),
             n_col=1,
             n_row=1,
         )
-    plot_gallery(
-        "%s - Train time %.1fs" % (name, train_time), components_[:n_components]
-    )
+
+    plot_gallery(name, components_[:n_components])
 
 plt.show()
+# %%
+# Decomposition: Dictionary learning
+# ---------------------------------------------------
+#
+# In the further section, let's consider :ref:`DictionaryLearning` more precisely.
+# Dictionary learning is a problem that amounts to finding a sparse representation
+# of the input data as a combination of simple elements. These simple elements form
+# a dictionary. It is possible to constrain the dictionary and/or code to be positive
+# to match constraints that may be present in the data.
 
-# #############################################################################
-# Various positivity constraints applied to dictionary learning.
-estimators = [
+dict_estimators = [
```

Comment:
Could you as well break the for loop into three different cells?

Reply:
Also done in the 43d4a5b.